### DISCUSSION TITLE
is it langium validation can show validation error  by default

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1427] [Q&A] [answered]
is it langium validation can show validation error  by default?i only add empty validatior(below StatemachineValidator),but not implement it,when i launch vscode extension,langium parse error and lexer error  message show in  example1.st file
![image](https://github.com/eclipse-langium/langium/assets/26995040/b571eff2-9004-410a-8f6a-9810e80c8271)
![image](https://github.com/eclipse-langium/langium/assets/26995040/22429fad-9afc-4cef-b34a-817f2c4bafbb)
![image](https://github.com/eclipse-langium/langium/assets/26995040/9df34619-0d61-4891-9e5a-c188e66aecaa)
![image](https://github.com/eclipse-langium/langium/assets/26995040/571dcfc0-6167-4a6b-9c60-1a46545647f6)

**Comment by zrwsmd** [comment]
i am considering using vscode.language.createDiagnosticCollection API and vscode.languages.registerCodeActionsProvider to collect langium parse and lexer error info ,and show diagnosticInfo in the example1.st  file,but langium seems to have implement it like below screenshot?

**Comment by msujew** [comment, marked as answer]
Langium sends diagnostics to the client using the LSP push diagnostics mechanism. By default, it will convert all parser and lexer errors into error diagnostics. In addition, you can use the semantic validation API, to perform custom validation based on your AST.

The `vscode-languageclient` package takes care of using the vscode API to display diagnostics received from the language server connection. No need to implement this yourself.
